src/libmongoc/tests/test-mongoc-dns.c

```diff
@@ -701,11 +699,9 @@ _prose_test_9 (bool pooled)
       _prose_loadbalanced_ping (client);
    }
 
-   bson_mutex_lock (&topology->tpld_modification_mtx);
    expected_hosts = MAKE_HOSTS ("localhost.test.build.10gen.cc:27017");
    check_topology_description (mc_tpld_unsafe_get_mutable (client->topology),
                                expected_hosts);
-   bson_mutex_unlock (&topology->tpld_modification_mtx);
```

Comment:
Possibly, although `check_topology_description` is expecting `add_server()` to be a no-op (because the server description should already be registered). But also, you are right that another thread managing the pool might modify it. This can be handled by just taking a `tpld_modification` and using that to restore thread-safety, even though we don't intend to modify the topology.